cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/migrations/MigrateDatabase100To200.java

```diff
@@ -0,0 +1,231 @@
+                    db.update("revs", lowestSequenceAsParent, "parent=?", duplicateSequenceArg);
+
+                    // If multipart was used it is possible attachments were not added to the lowest
+                    // and could be on any of the revisions, so we need to update them too.
+                    // This might create duplicates on the lowest sequence, but we handle duplicates
+                    // later anyway.
+                    db.update("attachments", lowestSequenceAsSequence, "sequence=?",
+                            duplicateSequenceArg);
+
+                    // Finally delete the duplicate revs
+                    db.delete("revs", "sequence=?", duplicateSequenceArg);
+                }
+
+                // For the lowest sequence we should clean up any duplicate attachments that may
+                // have resulted from a replication forceInsert using multi-part or our earlier
+                // migration of attachments from higher sequence duplicates to the lowest.
+                Map<String, Integer> duplicateAttachmentFilenames = new HashMap<String, Integer>();
+                c = db.rawQuery("SELECT filename, COUNT(*) FROM attachments WHERE sequence=? " +
+                        "GROUP BY " +
+                        "filename HAVING COUNT(*) > 1", new String[]{lowest.sequence.toString()});
+                while (c.moveToNext()) {
+                    duplicateAttachmentFilenames.put(c.getString(0), c.getInt(1));
+                }
+                c.close();
+                // Loop through the duplicate attachment entries
+                for (Map.Entry<String, Integer> duplicateAttachmentFilename :
+                        duplicateAttachmentFilenames.entrySet()) {
+                    String[] whereArgs = new String[]{lowest.sequence.toString(),
+                            duplicateAttachmentFilename.getKey(),
+                            Integer.toString(duplicateAttachmentFilename.getValue() - 1)};
+                    // Since the attachments should be identical it doesn't matter which entry we
+                    // delete, but we'll leave the first row.
+                    // Unfortunately DELETE ... LIMIT is generally not enabled in Android so we use
+                    // a subquery to get the rowids for the duplicates and limit to one less than
+                    // the total number of identical attachments we have so that we keep one.
+                    db.delete("attachments", "rowid IN (SELECT rowid FROM attachments WHERE " +
+                            "sequence=? AND filename=? ORDER BY rowid DESC limit ?)", whereArgs);
+                }
+
+                // Finally resolve the winner for the document based on all the merged branches
+                reevaluateWinner(db, lowest.doc_id);
+            }
+        } finally {
+            DatabaseUtils.closeCursorQuietly(c);
+        }
+    }
+
+    /**
+     * This method copies the code from
```

Comment:
Once https://github.com/cloudant/sync-android/pull/334/ is merged you can use the new `pickWinnerOfConflicts` instead.